lib/Controller/ChattyLLMController.php

```diff
@@ -0,0 +1,362 @@
+			$message->setRole($role);
+			$message->setContent($content);
+			$message->setTimestamp($timestamp);
+			$this->messageMapper->insert($message);
+			return new JSONResponse();
+		} catch (\OCP\DB\Exception $e) {
+			$this->logger->warning('Failed to add a chat message', ['exception' => $e]);
+			return new JSONResponse(['error' => $this->l10n->t('Failed to add a chat message')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param int $sessionId
+	 * @param int $limit
+	 * @param int $cursor
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function getMessages(int $sessionId, int $limit = 20, int $cursor = 0): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		try {
+			$sessionExists = $this->sessionMapper->exists($sessionId);
+			if (!$sessionExists) {
+				return new JSONResponse(['error' => $this->l10n->t('Session not found')], Http::STATUS_NOT_FOUND);
+			}
+
+			$messages = $this->messageMapper->getMessages($sessionId, $cursor, $limit);
+			if ($messages[0]->getRole() === $this->l10n->t('system')) {
+				array_shift($messages);
+			}
+
+			return new JSONResponse($messages);
+		} catch (\OCP\DB\Exception $e) {
+			$this->logger->warning('Failed to get chat messages', ['exception' => $e]);
+			return new JSONResponse(['error' => $this->l10n->t('Failed to get chat messages')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param integer $messageId
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function deleteMessage(int $messageId): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		try {
+			$this->messageMapper->deleteMessageById($messageId);
+			return new JSONResponse();
+		} catch (\OCP\DB\Exception | \RuntimeException $e) {
+			$this->logger->warning('Failed to delete a chat message', ['exception' => $e]);
+			return new JSONResponse(['error' => $this->l10n->t('Failed to delete a chat message')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param integer $sessionId
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function generate(int $sessionId): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		try {
+			$sessionExists = $this->sessionMapper->exists($sessionId);
+			if (!$sessionExists) {
+				return new JSONResponse(['error' => $this->l10n->t('Session not found')], Http::STATUS_NOT_FOUND);
+			}
+
+			$stichedPrompt =
+				$this->getStichedMessages($sessionId)
+				. PHP_EOL
+				. $this->l10n->t('assistant') . ': ';
+
+			$result = $this->queryLLM($stichedPrompt);
+
+			$message = new Message();
+			$message->setSessionId($sessionId);
+			$message->setRole($this->l10n->t('assistant'));
```

Comment:
It will be needed for the final prompt since there might be a model that does not understand English well or on the opposite end, might be biased towards English and might not reply in the asked language.

But we should set it in English in the db.

Reply:
following this (https://github.com/nextcloud/assistant/pull/86#discussion_r1598250791), it would make sense to leave the roles untranslated everywhere.